### ISSUE TITLE
Feature Request: shorter filename in output

### ISSUE THREAD

**Issue by laoshaw** [rs/zerolog#402]
By default zerolog outputs a full path and filename, sometimes it's just too long, can I have the filename directly instead of its full-path/filename? can I configure zerolog with some flags to do that?

**Comment by laoshaw** [comment]
Basically, the "Lshortfile" flag in stdlib's log package.

**Comment by weisdd** [comment]
@laoshaw zerolog has a global variable called `CallerMarshalFunc` that you can use to form whatever name you prefer:

https://github.com/rs/zerolog/blob/c0c2e11fc3cd04ae28d856789cf58ffd1666bc3f/globals.go#L63-L66

The standard implementation in Go log package looks like this ([Source](https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/log/log.go;drc=926994fd7cf65b2703552686965fb05569699897;l=134)):
```go
if l.flag&Lshortfile != 0 {
    short := file
    for i := len(file) - 1; i > 0; i-- {
        if file[i] == '/' {
            short = file[i+1:]
            break
        }
    }
    file = short
}
```


So, basically, here's the way to get something similar to "Lshortfile":

```go
zerolog.CallerMarshalFunc = func(file string, line int) string {
    short := file
    for i := len(file) - 1; i > 0; i-- {
        if file[i] == '/' {
            short = file[i+1:]
            break
        }
    }
    file = short
    return file + ":" + strconv.Itoa(line)
}
```

With that variable set, a call:

```go
zlog.Info().Caller().Msg("test")
```

Would produce:

```json
{"level":"info","caller":"main.go:90","time":"2022-03-12T15:56:42+01:00","message":"test"}
```

**Comment by laoshaw** [comment]
Thanks. That shall work. Any chance if this can  be added to code base.

**Comment by weisdd** [comment]
@laoshaw I'm just a random zerolog user, can't comment on that. :)

**Comment by stefan-cross** [comment]
Just to note, it looks like the `CallerMarshalFunc` func singature has changed but the above would still work using:

```
func(_ uintptr, file string, line int)
```

**Comment by cgons** [comment]
There's now a simple way to do this - documented in the README: https://github.com/rs/zerolog?tab=readme-ov-file#add-file-and-line-number-to-log

I think this issue can be closed.
